### PR TITLE
Iterator design pattern

### PR DESCRIPTION
**Pull request by andretai**
Hi, I've made changes for my task (iterator), should not affect other parts I think.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,3 +5,5 @@
 2. Project Structure -> Import the entire "lib" folder as module dependency
 
 3. Run "ContactManagementSoftware.java"
+
```
Comment (wing8169) [inline README.md]: :<

File: src/contactmanagementsoftware/MUI.java [hunk 2]
```diff
@@ -194,18 +194,32 @@ public void setDescription() {
...
         DefaultTableModel tableModel = (DefaultTableModel) jXTable1.getModel();
         tableModel.setRowCount(0);
-        ArrayList<Acquaintances> list;
-        try {
-            list = acquaintances.get(Categories.getSelectedIndex());
-        } catch (Exception e) {
-            return;
-        }
-        for (int i = 0; i < list.size(); i++) {
+//        ArrayList<Acquaintances> list;
```
Comment (wing8169) [inline src/contactmanagementsoftware/MUI.java]: Str8 remove commented code

File: src/contactmanagementsoftware/MUI.java [hunk 3]
```diff
@@ -194,18 +194,32 @@ public void setDescription() {
...
+        Iterator iter = getIterator();
+        int index = 0;
+        while(iter.hasNext()) {
+            Acquaintances item = (Acquaintances) iter.next();
             String[] data = new String[4];
-            data[0] = Integer.toString(i + 1);
-            data[1] = list.get(i).getName();
-            data[2] = list.get(i).getMobileNo();
-            data[3] = list.get(i).getEmail();
+            data[0] = Integer.toString(index+1);
```
Comment (wing8169) [inline src/contactmanagementsoftware/MUI.java]: not sure but do you need ++index?

data[0] = Integer.toString(++index);

File: src/contactmanagementsoftware/MUI.java [hunk 4]
```diff
@@ -1099,17 +1113,27 @@ public Iterator getIterator() {
         @Override
         public boolean hasNext() {
 
-            if (index < acquaintances.get(categoryIndex).size()) {
-                return true;
+//            if (index < acquaintances.get(Categories.getSelectedIndex()).size()) {
```
Comment (wing8169) [inline src/contactmanagementsoftware/MUI.java]: remove commented code

File: src/contactmanagementsoftware/MUI.java [hunk 5]
```diff
@@ -1099,17 +1113,27 @@ public Iterator getIterator() {
...
+//                return true;
+//            }
+//            return false;
+
+            try {
+                if (index < acquaintances.get(Categories.getSelectedIndex()).size()) {
+                    return true;
```
Comment (wing8169) [inline src/contactmanagementsoftware/MUI.java]: return index < acquaintances.get(Categories.getSelectedIndex()).size());

File: src/contactmanagementsoftware/MUI.java [hunk 6]
```diff
@@ -1099,17 +1113,27 @@ public Iterator getIterator() {
...
+                }
+                return false;
+            } catch (Exception e) {
+                System.out.println(e.);
```
Comment (wing8169) [inline src/contactmanagementsoftware/MUI.java]: e.printStackTrace();

**Review by wing8169 — CHANGES_REQUESTED** [review]

File: src/contactmanagementsoftware/MUI.java (hunk 6 above)
Comment (andretai) [reply src/contactmanagementsoftware/MUI.java]: okay

File: src/contactmanagementsoftware/MUI.java (hunk 5 above)
Comment (andretai) [reply src/contactmanagementsoftware/MUI.java]: okay

File: src/contactmanagementsoftware/MUI.java (hunk 4 above)
Comment (andretai) [reply src/contactmanagementsoftware/MUI.java]: okay

File: src/contactmanagementsoftware/MUI.java (hunk 3 above)
Comment (andretai) [reply src/contactmanagementsoftware/MUI.java]: yeah ++index xD

File: src/contactmanagementsoftware/MUI.java (hunk 2 above)
Comment (andretai) [reply src/contactmanagementsoftware/MUI.java]: okay

File: README.md (hunk 1 above)
Comment (andretai) [reply README.md]: xD

**Review by wing8169 — APPROVED** [review]

**Pull request merged**